inference-engine/thirdparty/clDNN/src/gpu/ocl_builder.cpp

```diff
@@ -25,6 +25,49 @@ namespace cldnn {
 namespace gpu {
 static constexpr auto INTEL_PLATFORM_VENDOR = "Intel(R) Corporation";
 
+static std::vector<cl::Device> getSubDevices(cl::Device& rootDevice) {
+    cl_uint maxSubDevices;
+    size_t maxSubDevicesSize;
+    const auto err = clGetDeviceInfo(rootDevice(),
+                                     CL_DEVICE_PARTITION_MAX_SUB_DEVICES,
+                                     sizeof(maxSubDevices),
+                                     &maxSubDevices, &maxSubDevicesSize);
+
+    if (err != CL_SUCCESS || maxSubDevicesSize != sizeof(maxSubDevices)) {
+        throw cl::Error(err, "clGetDeviceInfo(..., CL_DEVICE_PARTITION_MAX_SUB_DEVICES,...)");
+    }
+
+    if (maxSubDevices == 0) {
+        return {};
+    }
+
+    const auto partitionProperties = rootDevice.getInfo<CL_DEVICE_PARTITION_PROPERTIES>();
+    const auto partitionable = std::any_of(partitionProperties.begin(), partitionProperties.end(),
+                                            [](const decltype(partitionProperties)::value_type& prop) {
+                                                return prop == CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN;
+                                            });
+
+    if (!partitionable) {
+        return {};
+    }
+
+    const auto partitionAffinityDomain = rootDevice.getInfo<CL_DEVICE_PARTITION_AFFINITY_DOMAIN>();
+    const decltype(partitionAffinityDomain) expectedFlags =
+        CL_DEVICE_AFFINITY_DOMAIN_NUMA | CL_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE;
+
+    if ((partitionAffinityDomain & expectedFlags) != expectedFlags) {
+        return {};
+    }
```

Comment:
Can we just check that `partitionAffinityDomain != 0`? According specification CL_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE will use any available division (include CL_DEVICE_AFFINITY_DOMAIN_NUMA and other available levels), if the device split is not supported it returns 0, so looks like any `!= 0` number is acceptable for us

Reply:
Seems like your approach doesn't work.